### PR TITLE
Finished implementing moves between hands and piles 

### PR DESCRIPTION
**Pull request by amittovim**
please check the code.

### CODE REVIEW

File: src/app/enums/pile.enum.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 export const PileEnum = {
-    DrawPile: 'Draw Pile',
-    DiscardPile: 'Discard Pile',
+    DrawPile: 'drawPile',
+    DiscardPile: 'discardPile',
+    HumanHand: 'humanHandPile',
+    BotHand: 'botHandPile'
```
Comment (DorAm) [inline src/app/enums/pile.enum.js:5]: Why did you change these to lowercase?

File: src/app/game/board/board.component.js [hunk 2]
```diff
@@ -7,24 +7,42 @@ import DeckService from "./deck/deck.service";
...
+                      moveCardDriver1={this.moveCard}
+                />
             </div>
         );
     }
+
+    moveCard(card, sourcePile) {
+        this.props.boardService.moveCard(card, this.state[sourcePile.name]);
+        console.log(this.state);
+        this.forceUpdate(); // TODO: ask Offer how to avoid this?
```
Comment (DorAm) [inline src/app/game/board/board.component.js]: We should find a better way to do this. This is a bad practice like Offer wrote in his slides

File: src/app/game/board/board.service.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * as dealer from '../../../logic/dealer/dealer';
+import {GameState} from "../../../logic/state";
+
+export default class BoardService {
+
+    static moveCard(card, sourcePile) {
+        dealer.handleMoveCard(card, sourcePile);
+
+        // TODO: remove this from here. its temporary just to see that the moves are working
```
Comment (DorAm) [inline src/app/game/board/board.service.js:9]: I think you forgot to remove this

File: src/app/game/board/deck/deck.component.js [hunk 4]
```diff
@@ -11,26 +11,39 @@ class Deck extends Component {
             drawPile: this.props.drawPile
         };
 
-        this.moveCard = this.moveCard.bind(this);
+    /*    this.moveCard = this.moveCard.bind(this);*/
```
Comment (DorAm) [inline src/app/game/board/deck/deck.component.js]: please don't leave any commented out lines in code you want to push

File: src/app/game/board/deck/deck.component.js [hunk 5]
```diff
@@ -11,26 +11,39 @@ class Deck extends Component {
...
                           cards={this.state.drawPile.cards}
-                          onCardClick={this.moveCard} />
+                          // onCardClick={this.moveCard}
+                          moveCardDriver1={this.moveCardDriver1}
+                />
             </div>
         );
     }
 
+    moveCardDriver1(card, sourcePile) {
```
Comment (DorAm) [inline src/app/game/board/deck/deck.component.js]: I don't understand this. what is card driver 0\1 ?

File: src/app/game/board/deck/deck.service.js [hunk 6]
```diff
@@ -2,8 +2,10 @@ import * as dealer from '../../../../logic/dealer/dealer';
 
 export default class DeckService {
 
+/*
     static moveCard(card, sourcePile, destinationPile) {
         dealer.handleMoveCard(card, sourcePile, destinationPile);
     }
+*/
```
Comment (DorAm) [inline src/app/game/board/deck/deck.service.js]: please delete this chunk of comments

File: src/app/game/board/deck/discard-pile/discard-pile.component.js [hunk 7]
```diff
@@ -5,17 +5,26 @@ import Card from "../../../../shared/components/card/card.component";
...
     constructor(props) {
         super(props);
+        this.state = {
 
-        this.state = {}
+        }
+        this.moveCardDriver2 = this.moveCardDriver2.bind(this);
     }
 
+    moveCardDriver2(card) {
```
Comment (DorAm) [inline src/app/game/board/deck/discard-pile/discard-pile.component.js:14]: again , moveCardDriver2?

File: src/app/shared/components/card/card.component.js [hunk 8]
```diff
@@ -50,12 +55,16 @@ class Card extends Component {
...
+        this.props.moveCardDriver2(this.props.card);
+        // this.props.onCardClick(this.props.card);
         // this.setState(prevState => ({
         //     isOnBackSide: !this.state.isOnBackSide
         // }));
     };
 
+    flipCard() {
+        this.setState({isOnBackSide: !this.state.isOnBackSide})
+    }
```
Comment (DorAm) [inline src/app/shared/components/card/card.component.js]: I suggest we don't deel with flip card for now. Let's finish developing the game with visible cards, it will make our lives easier

File: src/logic/dealer/dealer.js [hunk 9]
```diff
@@ -39,11 +47,50 @@ function drawStartingCard() {
...
 export function handleMoveCard(card, sourcePile, destinationPile) {
     moveCard(card, sourcePile, destinationPile);
     setLeadingCard(card, destinationPile);
     // rollOverCard();
 }
+*/
+
+export function handleMoveCard(card, sourcePile) {
+    let destinationPile;
+    debugger;
```
Comment (DorAm) [inline src/logic/dealer/dealer.js]: remove debugger, last time it costed us 5 points

File: src/logic/init/draw-pile.init.js [hunk 10]
```diff
@@ -19,16 +19,17 @@ function createDrawPile() {
     GameState.drawPile = new PileModel(PileEnum.DrawPile);
     createNumberCards();
     createActionCards();
-    utils.shuffleArray(GameState.drawPile.cards);
+
+    //TODO : unremark the following line
+    //utils.shuffleArray(GameState.drawPile.cards);
```
Comment (DorAm) [inline src/logic/init/draw-pile.init.js]: remove comments

File: src/logic/init/draw-pile.init.js [hunk 11]
```diff
@@ -19,16 +19,17 @@ function createDrawPile() {
...
 }
 
 function createNumberCards() {
     for (const number in CardNumberEnum) {
-        if (number === CardNumberEnum.Two) {
-            continue;
-        }
+        // if (number === CardNumberEnum.Two) {
+        //     continue;
+        // }
```
Comment (DorAm) [inline src/logic/init/draw-pile.init.js]: remove comments

**Review by DorAm — CHANGES_REQUESTED** [review]
Please read my CR comments and change the code accordingly.
Thanks

File: src/app/enums/pile.enum.js (hunk 1 above)
Comment (amittovim) [reply src/app/enums/pile.enum.js:5]: i also added the two hand piles... 
i did it because i needed them to be equal to the fields of state in the "board.component.js" file

Comment (amittovim) [reply src/app/enums/pile.enum.js:5]: in lines 12-15 on "board.component.js"

File: src/app/game/board/board.service.js (hunk 3 above)
Comment (amittovim) [reply src/app/game/board/board.service.js:9]: we still need it. when we'll start the game flow we'll delete it.

File: src/app/game/board/deck/deck.component.js (hunk 4 above)
Comment (amittovim) [reply src/app/game/board/deck/deck.component.js]: ok, its just that im used to not wanting  to delete code unless you authorise it and i want you to see what code was there before.  but thats all the meaning of using GIT. :-) 
Eventually i will implement it .

File: src/app/game/board/deck/deck.component.js (hunk 5 above)
Comment (amittovim) [reply src/app/game/board/deck/deck.component.js]: we talked about it on the phone. and agreed to change the name if will keep it.

File: src/app/game/board/deck/deck.service.js (hunk 6 above)
Comment (amittovim) [reply src/app/game/board/deck/deck.service.js]: done

File: src/app/game/board/deck/discard-pile/discard-pile.component.js (hunk 7 above)
Comment (amittovim) [reply src/app/game/board/deck/discard-pile/discard-pile.component.js:14]: again, was explained on the phone

File: src/app/shared/components/card/card.component.js (hunk 8 above)
Comment (amittovim) [reply src/app/shared/components/card/card.component.js]: done

File: src/logic/dealer/dealer.js (hunk 9 above)
Comment (amittovim) [reply src/logic/dealer/dealer.js]: done

File: src/logic/init/draw-pile.init.js (hunk 11 above)
Comment (amittovim) [reply src/logic/init/draw-pile.init.js]: done

**Comment by amittovim** [comment]
Please check my answers and give me an OK to continue with the merge

File: src/app/enums/pile.enum.js (hunk 1 above)
Comment (DorAm) [reply src/app/enums/pile.enum.js:5]: ok

File: src/logic/init/draw-pile.init.js (hunk 10 above)
Comment (amittovim) [reply src/logic/init/draw-pile.init.js]: done

File: src/logic/init/draw-pile.init.js (hunk 11 above)
Comment (amittovim) [reply src/logic/init/draw-pile.init.js]: done

File: src/app/game/board/board.component.js (hunk 2 above)
Comment (amittovim) [reply src/app/game/board/board.component.js]: Done
